src/main/java/project/controller/LoginController.java

```diff
@@ -33,25 +33,31 @@ public LoginController(UserService userService, MockDataService mockDataService)
         this.mockDataService = mockDataService;
     }
 
+    /**
+     *
+     * @param accessToken our own, server generated access token
+     * @param fbAccessToken access token we get from facebook
+     * @param id fb user id of the user, which we also use as user id
+     * @param name full name of user
+     * @param firstName first name of user
+     * @return
+     */
     @RequestMapping(value = "/login", method = RequestMethod.POST)
     public ResponseEntity<LoginResponse> login(
                                         @RequestParam(value="accessToken") String accessToken,
+                                        @RequestParam(value="fb_access_token") String fbAccessToken,
```

Comment:
Updating the access token flow, so if the client doesn't have a server generated access token stored in the browser, the client can send the fb access token which the server will validate and if everything is correct, then the server will return the generated server access token so the client can store it in the browser.